File: backend/app/schemas/follow_up_message.py
from datetime import date, datetime
from typing import List, Optional

from pydantic import BaseModel


class FollowUpTemplateEntry(BaseModel):
    key: str
    label: str
    window_label: str
    window_status: str  # 'a_tiempo' | 'temprano' | 'atrasado'
    sent_at: Optional[datetime] = None
    text: str


class FollowUpPanelEntry(BaseModel):
    reservation_id: int
    reservation_number: str
    display_customer: str
    display_vehicle: str
    event_date: date
    days_to_event: int
    phone: Optional[str] = None
    current_key: Optional[str] = None
    templates: List[FollowUpTemplateEntry]


class MarkSentRequest(BaseModel):
    template_key: str

File: frontend/src/types/followUpMessage.ts
export type WindowStatus = 'a_tiempo' | 'temprano' | 'atrasado';

export interface FollowUpTemplateEntry {
  key: string;
  label: string;
  window_label: string;
  window_status: WindowStatus;
  sent_at: string | null;
  text: string;
}

export interface FollowUpPanelEntry {
  reservation_id: number;
  reservation_number: string;
  display_customer: string;
  display_vehicle: string;
  event_date: string;
  days_to_event: number;
  phone: string | null;
  whatsapp_username: string | null;
  current_key: string | null;
  last_sent_at: string | null;
  templates: FollowUpTemplateEntry[];
}
